bokehjs/src/lib/models/annotations/size_bar.ts

```diff
@@ -0,0 +1,409 @@
+          x_scale: this._major_scale,
+          y_scale: this._minor_scale,
+          ...plot_attrs,
+        })
+        break
+      }
+      case "vertical": {
+        this._size_bar = new InternalPlot({
+          width_policy: height == "max" ? "max" : "fit",
+          height_policy: width == "max" ? "max" : "fit",
+          frame_width: height == "max" ? undefined : height,
+          frame_height: width == "max" ? undefined : width,
+          right: [this._major_axis],
+          x_range: this._minor_range,
+          y_range: this._major_range,
+          x_scale: this._minor_scale,
+          y_scale: this._major_scale,
+          ...plot_attrs,
+        })
+        break
+      }
+    }
+  }
+
+  override get computed_elements(): ElementLike[] {
+    return [...super.computed_elements, this._size_bar]
+  }
+
+  override async lazy_initialize(): Promise<void> {
+    await super.lazy_initialize()
+    this._size_bar_view = this._element_views.get(this._size_bar) as ViewOf<Plot>
+  }
+
+  private _last_bbox = new BBox()
+
+  override update_layout(): void {
+    this.layout = this._size_bar_view.layout
+    this.layout.on_resize((outer) => {
+      if (!outer.equals(this._last_bbox)) {
+        this._last_bbox = outer
+        this.parent.request_layout(true)
+      }
+    })
+  }
+
+  get renderer(): GlyphRenderer<RadialGlyph> | null {
+    const {renderer} = this.model
+    if (renderer == "auto") {
+      const renderers = this.plot_view.model.renderers.filter((r): r is GlyphRenderer<RadialGlyph> => {
+        return r instanceof GlyphRenderer && r.glyph instanceof RadialGlyph
+      })
+      switch (renderers.length) {
+        case 0: {
+          logger.warn("can't find any radial glyph renderers")
+          return null
+        }
+        case 1: {
+          return renderers[0]
+        }
+        default: {
+          logger.warn("found multiple radial glyph renderers; choosing the first one")
+          return renderers[0]
+        }
+      }
+    } else {
+      return renderer
+    }
+  }
+
+  get glyph_view(): RadialGlyphView | null {
+    const {renderer} = this
+    if (renderer == null) {
+      return null
+    } else {
+      const rv = this.plot_view.views.get_one(renderer)
+      return rv.glyph_view as RadialGlyphView
+    }
+  }
+
+  protected _paint(_ctx: Context2d): void {
+    const {glyph_view} = this
+    if (glyph_view == null) {
+      return
+    }
+
+    const bounds = (() => {
+      const {bounds} = this.model
+      return bounds == "auto" ? [-Infinity, Infinity] as const : bounds
+    })()
+
+    const r_min = Math.max(uniforms.min(glyph_view.radius), bounds[0])
+    const r_max = Math.min(uniforms.max(glyph_view.radius), bounds[1])
+
+    const start = r_min
+    const end = r_max
+
+    const n_ticks = 5
+    const t = new AdaptiveTicker({desired_num_ticks: n_ticks})
+    const ticks = t.get_ticks(start, end, new Range1d(), NaN)
+    const radii = ticks.major
```

Comment:
1) We should handle the case where the smallest `glyph_view.radius` is zero.
2) We should maybe also handle the case where start and end is the same.
2) I think bounds provided by the user should always be used.
3) Is it possible to just provide one bound?

Reply:
For 1 and 2, a crude implementation could be:

```suggestion
    const r_min = Math.max(uniforms.min(glyph_view.radius), bounds[0])
    const r_max = Math.min(uniforms.max(glyph_view.radius), bounds[1])
    const equal = r_max == r_min
    const eps = 0.000001

    const start = equal ? Math.max(r_min - eps, 0) : r_min
    const end = equal ? r_max + eps : r_max

    const n_ticks = equal ? 1 : 5
    const t = new AdaptiveTicker({desired_num_ticks: n_ticks})
    const ticks = t.get_ticks(start == 0 ? end * eps : start, end, new Range1d(), NaN)
    const radii = ticks.major

```